src/argilla/client/feedback/utils/assignment.py

```diff
@@ -0,0 +1,190 @@
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+
+import random
+from collections import defaultdict
+from typing import Any, Dict, List, Union
+
+from argilla.client.users import User
+from argilla.client.workspaces import Workspace
+
+# DOCS: Step 1: login argilla/ get all users (list/dict)/ get records
+
+
+# Step 2: assign records
+def assign_records_to_groups(
+    groups: Dict[str, List[Any]], records: List[Any], overlap: int, shuffle: bool = True
+) -> Dict[str, Dict[str, Any]]:
+    """
+    Assign records to predefined groups with controlled overlap (for the groups) and optional shuffle. All members of the same group will annotate the same records.
+
+    Args:
+        groups: A dictionary where keys are group names and values are lists of users.
+        records: A list of records to be assigned.
+        overlap: The number of times each record is assigned to consecutive groups (0 for no overlap).
+        shuffle: If True, shuffle the records before assignment. Defaults to True.
+
+    Returns:
+        A dictionary where each key is a group and its value is another dictionary, which maps usernames to their respective assigned records.
+
+    Raises:
+        ValueError: If `overlap` is higher than the number of groups or negative.
+    """
+    if overlap < 0 or overlap >= len(groups.keys()):
+        raise ValueError("Overlap must be less than the number of groups and must not be negative.")
+
+    if shuffle:
+        random.shuffle(records)
+
+    assignments = {}
+    assignments_grouped = {}
+    group_names = list(groups.keys())
+    num_groups = len(group_names)
+
+    group_records = defaultdict(list)
+    for idx, record in enumerate(records):
+        for offset in range(overlap + 1):
+            group_index = (idx + offset) % num_groups
+            group_name = group_names[group_index]
+            group_records[group_name].append(record)
+
+    for group, users in groups.items():
+        for user in users:
+            assignments[user] = group_records[group]
+
+        assignments_grouped[group] = {user.username: assignments.get(user, []) for user in users}
+
+    return assignments_grouped
+
+
+def assign_records_individuals(
+    users: List[Any], records: List[Any], overlap: int, shuffle: bool = True
+) -> Dict[str, List[Any]]:
+    """
+    Assign records to users with controlled overlap and optional shuffle.
+
+    Args:
+        users: A list of user objects, each with a 'username' attribute.
+        records: A list of record objects to be assigned to users.
+        overlap: The number of times each record is assigned to consecutive users (0 for no overlap).
+        shuffle: If True, the records list will be shuffled before assignment. Defaults to True.
+
+    Returns:
+        A dictionary where keys are usernames and values are lists of assigned records.
+    """
+    if overlap < 0 or overlap >= len(users):
```

Comment:
I guess you may want to have 2 or 3 annotators giving responses to one record, so this could be greater than 1? But let me know if I'm misunderstanding what the function is doing here.

Reply:
you are right @nataliaElv, I thought wrongly about this.l